twitch.py

```diff
@@ -828,19 +828,40 @@ async def _run(self):
                 exclude = self.settings.exclude
                 priority = self.settings.priority
                 priority_only = self.settings.priority_only
+                priority_by_time = self.settings.priority_by_time
                 next_hour = datetime.now(timezone.utc) + timedelta(hours=1)
-                for campaign in self.inventory:
+                campaigns = self.inventory
+                campaigns.sort(key=lambda c: c.ends_at, reverse=False) # lets us give priority to campaigns ending soonest later on
+                max_priority = len(campaigns)
+                max_wanted_priority = max_priority + len(priorities)
```

Comment:
`max_priority`: is the highest number possible for **non-priority** games.
`max_wanted_priority`: is the highest possible number for priority games. **The lowest priority game is always greater than `max_priority`.** This keeps the two lists separated when merged into `self.wanted_games` array